python/ray/data/grouped_dataset.py

```diff
@@ -111,6 +112,19 @@ def aggregate(self, *aggs: AggregateFn) -> Dataset[U]:
         metadata = ray.get(metadata)
         return Dataset(BlockList(blocks, metadata), self._dataset._epoch)
 
+    def _aggregate_on(self, agg_cls: type, on: AggregateOnT, *args, **kwargs):
```

Comment:
seems also defined in dataset, can we unify them. In terms of dependency, it should be defined in grouped_dataset and used in dataset?

Reply:
I thought about this for a bit, didn't see an obvious way to do it. I'll try to find a way to do this!